staging/src/k8s.io/component-base/logs/json/register/register_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package register
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/component-base/config"
+	"k8s.io/component-base/logs"
+)
+
+func TestJSONFlag(t *testing.T) {
+	o := logs.NewOptions()
+	c := o.Config
+	fs := pflag.NewFlagSet("addflagstest", pflag.ContinueOnError)
+	output := bytes.Buffer{}
+	normalizeFunc := func(name string) string {
+		f := fs.GetNormalizeFunc()
+		return string(f(fs, name))
+	}
+	// The register package init function registers the JSON logging format
+	// and the value of formats should be: "json", "text"
+	formats := fmt.Sprintf(`"%s"`, strings.Join(logs.LogRegistry.List(), `", "`))
```

Comment:
Could you try not to copy logic from `Options.AddFlags`, but try to use this function? Those thests are worthless if they are not integrating with `Options`.

Reply:
Ok, Updated.
Originally, I just wanted to focus on the --logging-format  flag.